File: packages/shared-types/src/tap.ts
export interface BrewTap {
  name: string;
  official: boolean;
}

export interface TapActionRequest {
  name: string;
}

export interface TapActionResult {
  name: string;
  action: "tap" | "untap";
  success: boolean;
  stdout?: string;
  stderr?: string;
}

File: Sources/BrewweryCore/Models/Tap.swift
import Foundation

public struct BrewTap: Identifiable, Hashable, Codable, Sendable {
    public var name: String
    /// `true` when the tap name starts with `homebrew/`.
    public var official: Bool

    public var id: String { name }

    public init(name: String, official: Bool) {
        self.name = name
        self.official = official
    }
}

public enum TapAction: String, Codable, Sendable, Hashable {
    case tap
    case untap
}
